### ISSUE TITLE
[question] How to format a cell value during CSV export to be in a different format than it's displayed in the grid?

### ISSUE THREAD

**Issue by dimon82** [mui/mui-x#14723]
### The problem in depth

I have a date column that displays a value in the format of 'MM/DD/YYYY', but during CSV export I want it to be 'MM/DD/YYYY h:mm:ss a' (include time). Is there a way to do it?

Also long numbers that are actually strings are getting exported with scientific notation like "2.5E+15", any way to make sure those long numbers are exported as strings instead?

**Search keywords**: DataGrid CSV export cell format
**Order ID**: corporate order (don't know it)

**Comment by MBilalShafi** [comment]
Your request seems similar to #8193. While a proper fix would be done with #8193 you can work around this by using a combination of the prop `ignoreValueFormatterDuringExport` and `valueFormatter`.
The trick is to set the values you want to export as actual values and use `valueFormatter` to show different values in the Data Grid.

Here's an example: https://stackblitz.com/edit/react-bv7urx?file=Demo.tsx

Let me know if it solves your issue.

**Comment by dimon82** [comment]
@MBilalShafi thank you for the reply. However, this solution won't really work for what I was trying to accomplish. Meaning, in my case I had to specify the type of the column as `date`  so I can use filters on the date only (without time). If I go with your fix, then csv will show it in `YYYY-MM-DDTHH:mm:ssZ` format which is not what I want. I want it to be in this format `'MM/DD/YYYY h:mm:ss a'`.

I ended up doing this:
```
renderCell: (params) => moment(params.value).format('MM/DD/YYYY'),
valueFormatter: (value) => moment(value).format('MM/DD/YYYY, h:mm:ss a')
```

not the solution I like as `renderCell` is not supposed to be used for this type of thing, but it does the trick. Grid uses `renderCell` to render value I want (as it seems like it has higher priority than `valueFormatter`) and `valueFormatter` in this case is used for csv export. I had to add a comma in there otherwise excel will format the date in its own way (such as `9/20/2024 18:19`, and again, why it doesn't include seconds and uses military time I have no idea). Not sure what the right way is to make sure that csv shows it as 'MM/DD/YYYY h:mm:ss a' (without comma).
